src/main/java/me/itzg/tsdbcassandra/config/DownsampleProperties.java

```diff
@@ -0,0 +1,79 @@
+package me.itzg.tsdbcassandra.config;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.convert.DurationFormat;
+import org.springframework.boot.convert.DurationStyle;
+import org.springframework.boot.convert.DurationUnit;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+@ConfigurationProperties("app.downsample")
+@Component
+@Data
+@Validated
+public class DownsampleProperties {
+  /**
+   * Specifies how many partitions should be used when tracking ingested data for downsampling.
+   * A value of zero disables downsample tracking.
+   */
+  @Min(0)
+  int partitions;
+
+  public boolean isTrackingEnabled() {
+    return partitions > 0;
+  }
+
+  @DurationUnit(ChronoUnit.HOURS)
+  Duration timeSlotWidth = Duration.ofHours(1);
+
+  /**
+   * The amount of time to allow for the lastTouch of pending downsample sets to remain stable.
+   * This is mainly useful for handling backfilled data where this delay ensures that all metrics
+   * have likely been ingested for the slot.
+   */
+  @DurationUnit(ChronoUnit.MINUTES)
+  Duration lastTouchDelay = Duration.ofMinutes(5);
+
+  /**
+   * Specifies how often pending downsample sets will be retrieved and processed.
+   */
+  @DurationUnit(ChronoUnit.MINUTES)
+  Duration downsampleProcessPeriod = Duration.ofMinutes(1);
+
+  /**
+   * Comma separated list of partitions or ranges of partitions, such as "0,5-8,12,15-18"
+   */
+  IntegerSet partitionsToProcess;
```

Comment:
So is this getting implicitly converted from a string because you have defined the IntegerSetConverter?  I have never noticed that bit of spring magic before.

Reply:
You are correct. This bit of Spring magic was actually harder to achieve than I originally thought -- originally I was wanting to use just a `Set<Integer>`; however, the pre-defined conversions were intercepting my own conversion. So, I ended up introducing a new data holder type to cleanly leverage the `@ConfigurationPropertiesBinding` support [documented here](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-conversion).